apis/logging/v1/index_management_types.go

```diff
@@ -47,8 +47,25 @@ type IndexManagementPhasesSpec struct {
 // +k8s:openapi-gen=true
 type IndexManagementDeletePhaseSpec struct {
 	// The minimum age of an index before it should be deleted (e.g. 10d)
-	//
 	MinAge TimeUnit `json:"minAge"`
+	// The per namesapce specification to delete documents older than a given minimum age
+	Namespaces []IndexManagementDeleteNamespaceSpec `json:"namespaceSpec"`
+
+	// If MinAge for NamespaceSpec is empty, delete the records matching the namespaces which are older than this default age
+	DefaultAge TimeUnit `json:"defaultAge"`
+}
+
+type IndexManagementDeleteNamespaceSpec struct {
+	// The unique name of the spec
+	// +optional
+	Name string `json:"name"` // delete-<namespace_name>
```

Comment:
This shouldn't be marked as `optional`. A unique name for each spec is required to be provided.
```suggestion
	// +required
	Name string `json:"name"` // delete-<namespace_name>
```